src/Symfony/Bundle/FrameworkBundle/Routing/Router.php

```diff
@@ -51,7 +51,11 @@ public function __construct(ContainerInterface $container, $resource, array $opt
     public function getRouteCollection()
     {
         if (null === $this->collection) {
+            $this->collection = new RouteCollection();
```

Comment:
The getter does not change the state of the property if an exception is thrown. But NOW it does. The next execution will not throw an exception anymore. So this makes it indeterministic from the outside. Again, if you go this its like doing

```
public function foobar() {
   if (0 === rand(0, 1)) { throw new Exception(); }
}
```